include/gridtools/common/hugepage_alloc.hpp

```diff
@@ -11,25 +11,178 @@
 #pragma once
 
 #include <atomic>
+#include <cassert>
 #include <cstdlib>
+#include <cstring>
 #include <new>
+#include <stdexcept>
+#if __cpp_lib_int_pow2 >= 202002L
+#include <bit>
+#endif
+#ifdef __linux__
+#include <cstdio>
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
 
 namespace gridtools {
+    namespace hugepage_alloc_impl_ {
+        inline std::size_t ilog2(std::size_t i) {
+#if __cpp_lib_int_pow2 >= 202002L
+            return std::bit_width(t) - 1;
+#else
+            std::size_t log = 0;
+            while (i >>= 1)
+                ++log;
+            return log;
+#endif
+        }
+
+#ifdef __linux__
```

Comment:
If all of the `#ifdef _linux_ .. #endif` blocks are rolled into a single block, instead of interleaving them throughout the code, it will help to improve readability of the code.

Reply:
Done.